src/cached_object_store/ghost.rs

```diff
@@ -0,0 +1,196 @@
+use lru::LruCache;
+
+use std::fmt::Debug;
+use std::hash::Hash;
+use std::num::NonZeroUsize;
+
+/// The weighter is used to calculate the weight of the cache entry.
+pub trait Weighter<K, V>: Fn(&K, &V) -> usize + Send + Sync + 'static {}
+impl<K, V, T> Weighter<K, V> for T where for<'a> T: Fn(&K, &V) -> usize + Send + Sync + 'static {}
+
+/// A simple wrapper of GhostQueue based on LRU eviction
+pub struct GhostQueue<K: Hash + Eq, V> {
```

Comment:
does this generic parameter considered as a must? i suppose ghost queue is expected to be used inside this cached object store, imho a concerete type might be simpler

Reply:
Considering that K and V are not yet sure (whether the location hash value is used as a key or direct Path type, and whether ObjectMeta is needed), I think that if there is a need to modify it later, generic will be much more convenient.

Reply:
will this generic be used in multiple places with different types? if it's not, i feel using concrete types is enough. if the type is likely to be changed in the future, we can change it inside the struct.

Reply:
that's fine. I personally love the idea of 'principle of least power'. if we can resolve a problem, it's mostly better to use the least power to make it possible, and we should not be afraid of possible changes in the future. i love this article a lot, it describes some practises in scala, but i guess many principles also apply in rust, which is also considered as a language with powerful expressiveness, just for reference: https://www.lihaoyi.com/post/StrategicScalaStylePrincipleofLeastPower.html